pkg/storageincentives/agent.go

```diff
@@ -298,8 +295,24 @@ func (a *Agent) claim(ctx context.Context) error {
 	}
 
 	if isWinner {
+		a.nodeState.SetLastWonRound(round)
+		a.nodeState.SetState(winner)
 		a.metrics.Winner.Inc()
+		err := a.nodeState.SetBalance(ctx)
+		if err != nil {
+			a.logger.Info("could not set balance", "err", err)
+		}
+
 		err = a.contract.Claim(ctx)
+		if err != nil {
+			a.logger.Info("calculate winner reward", "err", err)
+		}
+		err = a.nodeState.CalculateWinnerReward(ctx)
+		if err != nil {
+			a.logger.Info("calculate winner reward", "err", err)
+		}
+		a.nodeState.AddFee()
```

Comment:
you are missing SaveStatus calls in many places...
i think you only need to save status when you win, get frozen, or you play in a round

Reply:
@istae so `AddFee()` already has `saveStatus()`

https://github.com/ethersphere/bee/blob/a715eed2f81eed6d49164bae2a9606004f599261/pkg/storageincentives/nodestatus.go#L116

Reply:
If you look into node status package. You will see SaveStatus is also being called in some functions.